app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/newdevicenotice/NewDeviceNoticeEmailAccessScreen.kt

```diff
@@ -0,0 +1,199 @@
+import com.x8bit.bitwarden.ui.platform.base.util.EventsEffect
+import com.x8bit.bitwarden.ui.platform.base.util.createAnnotatedString
+import com.x8bit.bitwarden.ui.platform.base.util.standardHorizontalMargin
+import com.x8bit.bitwarden.ui.platform.components.button.BitwardenFilledButton
+import com.x8bit.bitwarden.ui.platform.components.scaffold.BitwardenScaffold
+import com.x8bit.bitwarden.ui.platform.components.toggle.BitwardenSwitch
+import com.x8bit.bitwarden.ui.platform.components.util.rememberVectorPainter
+import com.x8bit.bitwarden.ui.platform.theme.BitwardenTheme
+
+/**
+ * The top level composable for the new device notice email access screen.
+ */
+@Composable
+fun NewDeviceNoticeEmailAccessScreen(
+    onNavigateToTwoFactorOptions: () -> Unit,
+    viewModel: NewDeviceNoticeEmailAccessViewModel = hiltViewModel(),
+) {
+    val state by viewModel.stateFlow.collectAsStateWithLifecycle()
+    EventsEffect(viewModel = viewModel) { event ->
+        when (event) {
+            NavigateToTwoFactorOptions -> onNavigateToTwoFactorOptions()
+        }
+    }
+
+    BitwardenScaffold {
+        NewDeviceNoticeEmailAccessContent(
+            email = state.email,
+            isEmailAccessEnabled = state.isEmailAccessEnabled,
+            onEmailAccessToggleChanged = remember(viewModel) {
+                { newState ->
+                    viewModel.trySendAction(EmailAccessToggle(newState = newState))
+                }
+            },
+            onContinueClick = { viewModel.trySendAction(ContinueClick) },
+            modifier = Modifier,
+        )
+    }
+}
+
+@Composable
+private fun NewDeviceNoticeEmailAccessContent(
+    email: String,
+    isEmailAccessEnabled: Boolean,
+    onEmailAccessToggleChanged: (Boolean) -> Unit,
+    onContinueClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = modifier
+            .standardHorizontalMargin()
+            .fillMaxSize()
+            .verticalScroll(state = rememberScrollState()),
+    ) {
+        Spacer(modifier = Modifier.height(104.dp))
+        HeaderContent()
+        Spacer(modifier = Modifier.height(24.dp))
+        MainContent(
+            email = email,
+            isEmailAccessEnabled = isEmailAccessEnabled,
+            onEmailAccessToggleChanged = onEmailAccessToggleChanged,
+            modifier = Modifier,
+        )
+        Spacer(modifier = Modifier.height(24.dp))
+        BitwardenFilledButton(
+            label = stringResource(R.string.continue_text),
+            onClick = onContinueClick,
+            modifier = Modifier
+                .fillMaxSize()
+                .imePadding(),
+        )
+        Spacer(modifier = Modifier.navigationBarsPadding())
+    }
+}
+
+/**
+ * Header content containing the warning icon and title.
+ */
+
+@Suppress("MaxLineLength")
+@Composable
+private fun HeaderContent(
+    modifier: Modifier = Modifier,
+) {
```

Comment:
🎨 If you scope this it will inherit the `modifier` and other context from its parent Column scope. That would mean you don't need a wrapping `Column` within `HeaderContent` to set the `modifier` and `horizontalAlignment`. It would look like...

```kotlin
fun NewDeviceNoticeEmailAccessContent(...) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        modifier = modifier
            .standardHorizontalMargin()
            .fillMaxSize()
            .verticalScroll(state = rememberScrollState()),
    ) {
        Spacer(...)
        HeaderContent()
        // other components
    }
}

private fun ColumnScope.HeaderContent() {
    Image(...)
    Spacer(...)
    // other components
} 
```

This works because `HeaderContent` is using the same `modifier` and `horizontalAlignment` from the parent Column scope. It doesn't work quite as cleanly for `MainContent` because you're not using the same horizontal alignment as its parent Column scope.

Consider this more of a FYI than an actual change request since there's nothing technically wrong with the way you've done it.

Reply:
Oh I got it and thanks for the explanation, it makes sense specially the part that it's the same as the parent. I'll change it since it clears a lot of unnecessary code lines 👍